src/cockpit/samples.py

```diff
@@ -121,21 +126,24 @@ def detect_cpu_sensors(self, hwmonid: int, name: str) -> None:
                 if name != 'cpu_thermal':
                     continue
 
-            self.sensors.append(sensor_path)
+            self.sensors.append(entry.path)
 
-    def sample(self, samples: Samples) -> None:
+    def sample(self, samples: Samples, hwmon_path='/sys/class/hwmon') -> None:
         cpu_names = ['coretemp', 'cpu_thermal', 'k8temp', 'k10temp', 'atk0110']
 
         if not self.sensors:
-            # TODO: 2 ** 32?
-            for index in range(0, 2 ** 32):
+            for path in os.listdir(hwmon_path):
                 try:
-                    with open(f'/sys/class/hwmon/hwmon{index}/name') as f:
+                    sensor_path = os.path.join(hwmon_path, path)
+                    with open(os.path.join(sensor_path, 'name')) as f:
                         name = f.read().strip()
-                    if name in cpu_names:
-                        self.detect_cpu_sensors(index, name)
+
+                    if name not in cpu_names:
+                        continue
+
+                    self.detect_cpu_sensors(sensor_path, name)
                 except FileNotFoundError:
-                    break
+                    continue
```

Comment:
Is it still an expected case to see `FileNotFoundError` here?  We just iterated and found this, no?  Is this in case it hotswaps out in realtime?

Reply:
This is in case /sys/class/hwmon/hwmon1/name does not exists. Not sure if this can happen, but let's be safe.

`listdir` just iterates over /sys/class/hwmon/ so has no idea if `hwmon1/name` exists.

Reply:
gotcha.